community/kernel/src/main/java/org/neo4j/kernel/NeoStoreDataSource.java

```diff
@@ -828,6 +823,9 @@ private Procedures setupProcedures() throws KernelException, IOException
         procedures.registerComponent( GraphDatabaseService.class,
                 (ctx) -> dependencyResolver.resolveDependency( GraphDatabaseService.class ) );
 
+        InternalProcedureLoggingService loggingService = new InternalProcedureLoggingService( log.getUserLog( Procedures.class  ) );
+        procedures.registerComponent( LoggingService.class, (ctx) -> loggingService );
```

Comment:
Aren't Log and Logger internal? This was meant as just a public facade interface for use by procedure authors.
Is org.neo4j.logging part of the public api, if so, we could just hand down that, perhaps just masking out bulk logging as abusing that  may have undesirable properties.